### PR TITLE
Add Sdkcontext

### PR DESCRIPTION
**Pull request by umutunsal**

### CODE REVIEW

File: hive_agent/config/config.py [hunk 1]
```diff
@@ -33,7 +33,3 @@ def save_config(self):
         """Saves the current configuration state to disk."""
         with open(self.config_path, "w") as f:
             toml.dump(self.config, f)
-
-    def get_log_level(self):
```
Comment (ToJen) [inline hive_agent/config/config.py:37]: can you double check that this function is being used in some places? it should be sued

**Review by ToJen — CHANGES_REQUESTED** [review]

File: hive_agent/config/config.py (hunk 1 above)
Comment (umutunsal) [reply hive_agent/config/config.py:37]: Get log level method only used while creating new HiveAgent object and I've replaced it with load_default_config method in sdk_context. It still uses old config class and get the keys the same way with looking some keys in config and if cant find the key gets default values. 

https://github.com/umutunsal/hive-agent-py/blob/4f37573ccb69671849e97344f204d8630b6772b8/hive_agent/sdk_context.py#L28

File: tests/database/test_database.py [hunk 2]
```diff
@@ -14,20 +14,21 @@
...
+        # await db_manager.create_table("users", columns)
+        # mock_session.add.assert_called_once()
+        # mock_session.commit.assert_called_once()
 
-        # Test invalid columns
-        with self.assertRaises(ValueError):
-            await db_manager.create_table("users", "invalid_columns")
+        # # Test invalid columns
+        # with self.assertRaises(ValueError):
+        #     await db_manager.create_table("users", "invalid_columns")
```
[the remark this answers is not in the document]
Comment (umutunsal) [reply tests/database/test_database.py]: Updated whole test_database class

File: hive_agent/llm_settings.py [hunk 3]
```diff
@@ -19,27 +19,27 @@
 
 
 def init_llm_settings(config):
-    model = config.get("model", "model", "gpt-3.5-turbo")
```
[the remark this answers is not in the document]
Comment (umutunsal) [reply hive_agent/llm_settings.py:22]: There is no additional update on toml file. We are still looking the same keys while creating default_config in sdk_context

File: hive_agent/database/database.py [hunk 4]
```diff
@@ -104,6 +111,11 @@ async def get_table_definition(self, table_name: str):
                 select(TableDefinition).filter_by(table_name=table_name)
             )
             table_definition = result.scalars().first()
+
+            # while running test
+            # scalars_result = await result.scalars()
+            # table_definition = await scalars_result.first()
```
[the remark this answers is not in the document]
Comment (umutunsal) [reply hive_agent/database/database.py]: Removed not necessary.

File: hive_agent/agent.py [hunk 5]
```diff
@@ -44,6 +44,19 @@ class HiveAgent:
     wallet_store: "WalletStore"
     __agent: Any
 
+    _instance = None
```
Comment (ToJen) [inline hive_agent/agent.py:47]: what's the instance for

File: hive_agent/server/routes/files.py [hunk 6]
```diff
@@ -55,6 +56,7 @@ async def create_upload_files(files: List[UploadFile] = File(...)):
                     logger.info("Inserting data to existing basic index")
                     logger.info(f"Index: {index_store.list_indexes()}")
                     index_store.save_to_file()
+                    hive_agent.recreate_agent()
```
Comment (ToJen) [inline hive_agent/server/routes/files.py:59]: is this function only used in tests

Comment (umutunsal) [reply hive_agent/server/routes/files.py:59]: it is used both on hive-agent-py/hive_agent/server/routes/files.py and  hive-agent-py/tests/test_agent.py.

Comment (umutunsal) [reply hive_agent/server/routes/files.py:59]: We need to reinit the agent with new tools and tool_retriever object

**Pull request closed without merging**